src/context-providers/config/form-builder-config.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { createContext, ReactNode, useContext } from "react";
+import { IFormBuilderConfig } from "./types";
+
+const FormBuilderConfigContext = createContext<IFormBuilderConfig>({});
+
+export const FormBuilderConfigProvider: React.FC<{
+    children: ReactNode;
+    value: IFormBuilderConfig;
+}> = ({ children, value }) => {
+    return (
+        <FormBuilderConfigContext.Provider value={value || {}}>
+            {children}
+        </FormBuilderConfigContext.Provider>
+    );
+};
+
+export const useFormBuilderConfigContext = () => {
+    const context = useContext(FormBuilderConfigContext);
+    if (context === undefined) {
+        throw new Error(
+            "useFormBuilderConfigContext must be used within a FormBuilderConfigContext"
+        );
+    }
+    return context;
+};
```

Comment:
- I foresee the hook to get bigger so move it to a separate hook file
- why the need to check for the presence of the context? under what circumstances will the hook be used outside the context provider?